src/provider/provider_file_memory.c

```diff
@@ -397,16 +397,21 @@ static umf_result_t file_alloc(void *provider, size_t size, size_t alignment,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    // alignment must be a power of two and a multiple of sizeof(void *)
-    if (alignment &&
-        ((alignment & (alignment - 1)) || (alignment % sizeof(void *)))) {
-        LOG_ERR("wrong alignment: %zu (not a power of 2 or a multiple of "
-                "sizeof(void *))",
-                alignment);
+    file_memory_provider_t *file_provider = (file_memory_provider_t *)provider;
+
+    // alignment must be a power of two and a multiple or a divider of the page size
+    if (alignment && ((alignment & (alignment - 1)) ||
+                      ((alignment % file_provider->page_size) &&
+                       (file_provider->page_size % alignment)))) {
```

Comment:
Just to make sure that I understand it right, with file provider we do not have the issue like with the devdax provider because the page size is 4KB in case of file provider. Right?

Reply:
Do you mean issue that size is not a multiplier of the page size?

Reply:
Yes, I updated the code.

Reply:
> Do you mean issue that size is not a multiplier of the page size?

yes